nav2_ceres_costaware_smoother/include/nav2_ceres_costaware_smoother/smoother_cost_function.hpp

```diff
@@ -0,0 +1,252 @@
+   * @param pt Point Xi for evaluation
+   * @param pt_p Point Xi+1 for calculating Xi's cost
+   * @param pt_m Point Xi-1 for calculating Xi's cost
+   * @param r Residual (cost) of term
+   */
+  template<typename T>
+  inline void addSmoothingResidual(
+    const double & weight,
+    const Eigen::Matrix<T, 2, 1> & pt,
+    const Eigen::Matrix<T, 2, 1> & pt_p,
+    const Eigen::Matrix<T, 2, 1> & pt_m,
+    T & r) const
+  {
+    Eigen::Matrix<T, 2, 1> d_pt_p = pt_p - pt;
+    Eigen::Matrix<T, 2, 1> d_pt_m = pt - pt_m;
+    Eigen::Matrix<T, 2, 1> d_pt_diff = _next_to_last_length_ratio * d_pt_p - d_pt_m;
+    r += (T)weight * d_pt_diff.dot(d_pt_diff);    // objective function value
+  }
+
+  /**
+   * @brief Cost function term for maximum curved paths
+   * @param weight Weight to apply to function
+   * @param pt Point Xi for evaluation
+   * @param pt_p Point Xi+1 for calculating Xi's cost
+   * @param pt_m Point Xi-1 for calculating Xi's cost
+   * @param curvature_params A struct to cache computations for the jacobian to use
+   * @param r Residual (cost) of term
+   */
+  template<typename T>
+  inline void addCurvatureResidual(
+    const double & weight,
+    const Eigen::Matrix<T, 2, 1> & pt,
+    const Eigen::Matrix<T, 2, 1> & pt_p,
+    const Eigen::Matrix<T, 2, 1> & pt_m,
+    T & r) const
+  {
+    Eigen::Matrix<T, 2, 1> center = arcCenter(
+      pt_m, pt, pt_p,
+      _next_to_last_length_ratio < 0 ? -1 : 1);
+    if (ceres::IsInfinite(center[0])) {
+      return;
+    }
+    T turning_rad = (pt - center).norm();
+    T ki_minus_kmax = (T)1.0 / turning_rad - _params.max_curvature;
+
+    if (ki_minus_kmax <= (T)EPSILON) {
+      return;
+    }
+
+    r += (T)weight * ki_minus_kmax * ki_minus_kmax;  // objective function value
+  }
+
+  /**
+   * @brief Cost function derivative term for steering away changes in pose
+   * @param weight Weight to apply to function
+   * @param xi Point Xi for evaluation
+   * @param xi_original original point Xi for evaluation
+   * @param r Residual (cost) of term
+   */
+  template<typename T>
+  inline void addDistanceResidual(
+    const double & weight,
+    const Eigen::Matrix<T, 2, 1> & xi,
+    const Eigen::Matrix<T, 2, 1> & xi_original,
+    T & r) const
+  {
+    r += (T)weight * (xi - xi_original).squaredNorm();  // objective function value
+  }
+
+  /**
+   * @brief Cost function term for steering away from costs
+   * @param weight Weight to apply to function
+   * @param value Point Xi's cost'
+   * @param params computed values to reduce overhead
+   * @param r Residual (cost) of term
+   */
+  template<typename T>
+  inline void addCostResidual(
+    const double & weight,
+    const Eigen::Matrix<T, 2, 1> & pt,
+    const Eigen::Matrix<T, 2, 1> & pt_p1,
+    const Eigen::Matrix<T, 2, 1> & pt_m1,
+    T & r) const
+  {
+    if (_params.cost_check_points.empty()) {
+      Eigen::Matrix<T, 2,
+        1> interp_pos = (pt - _costmap_origin.template cast<T>()) / (T)_costmap_resolution;
```

Comment:
The line breaks here are odd, why not keep all the type on 1 line and use the line extension after the `=`?

Reply:
this was done by uncrustify

Reply:
Sometimes it picks odd locations to make the break. It would also be acceptable by the formatting to do `Eigen::Matrix<T, 2,1> interp_pos =` on one line and have the rest be the line extension. It doesn't do well with templates / commas for breaking lines up.